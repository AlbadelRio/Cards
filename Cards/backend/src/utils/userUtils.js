function dataChanges(body, userData, userFound) {
  switch (body.type) {
    case 'cardsAdquired':
      userData.cardsAdquired.push(body.cardId);
      userFound.save();
      break;

    case 'difficultCards':
      userData.difficultCards.push(body.cardId);
      userFound.save();
      break;
    default:
      break;
  }
}

module.exports = { dataChanges };
