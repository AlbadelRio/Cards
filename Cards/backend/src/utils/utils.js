function DataChanges(body, userData, userFound) {
  switch (body.type) {
    case 'cardsAdquired':
      userData.cardsAdquired.push(body.cardId);
      userFound.save();
      break;

    case 'difficultCards':
      userData.difficultCards.push(body.cardId);
      userFound.save();
      break;
    case 'stats': {
      const existingDate = userData.stats.find((data) => data.date === body.date);
      if (!existingDate) {
        userData.stats.push(body);
        userFound.save();
      } else {
        existingDate.stats.update(
          {
            $push: { timePlayed: body.time, percentages: body.percentage },
            $sort: { score: -1 }
          }

        );
      } }
      break;
    default:
      break;
  }
}

module.exports = { DataChanges };
