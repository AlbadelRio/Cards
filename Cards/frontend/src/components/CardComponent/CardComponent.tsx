/* eslint-disable no-underscore-dangle */
/* eslint-disable react/destructuring-assignment */
import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {
  View, TextInput, TouchableOpacity, Image, StyleSheet
} from 'react-native';
import { updateCard } from '../../redux/actions/cardActionCreator';

const styles = StyleSheet.create({
  tinyLogo: {
    width: 30,
    height: 30
  },
  items: {
    width: 400,
    height: 300
  }
});

export default function Card({ card }:any) {
  const [question, setQuestion] = useState(card.question);
  const [answer, setAnswer] = useState(card.answer);
  const { token, refreshToken } = useSelector((store:any) => store.tokensReducer);
  const dispatch = useDispatch();
  const body = {
    question,
    answer

  };

  function handleUpdate() {
    if (question || answer) {
      dispatch(updateCard(
        token,
        refreshToken,
        card._id,
        body
      ));
    }
  }

  return (
    <View
      style={styles.items}
    >

      <View>
        <TextInput
          key={card._id}
          value={question}
          onChangeText={(text:any) => setQuestion(text)}
        />
        <TextInput
          key={card._id}
          value={answer}
          onChangeText={(text:any) => setAnswer(text)}
        />
      </View>
      <TouchableOpacity
        onPress={handleUpdate}
      >
        <View>
          <Image
            style={styles.tinyLogo}
            source={{
              uri: 'https://img.icons8.com/material-sharp/24/000000/edit--v3.png'
            }}
          />
        </View>
      </TouchableOpacity>
    </View>

  );
}
