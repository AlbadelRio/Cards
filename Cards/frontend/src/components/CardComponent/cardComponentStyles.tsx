import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  tinyLogo: {
    width: 30,
    height: 30
  },
  items: {
    width: 80,
    height: 80
  },

  card: {
    width: 350,
    height: 200,
    borderRadius: 3,
    backgroundColor: '#EAECF2',
    justifyContent: 'center',
    padding: 10,
    marginTop: 60,
    elevation: 10
  },
  modify: {
    alignItems: 'flex-end'
  }

});
