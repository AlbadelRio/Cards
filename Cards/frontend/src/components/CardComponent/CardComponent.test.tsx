import React from 'react';
import { render } from '../../utils/test.utils';
import CardComponent from './CardComponent';

jest.mock('../../redux/actions/cardActionCreator', () => ({
  ...jest.requireActual('../../redux/actions/cardActionCreator'),
  updateCard: jest.fn()
}));

describe('Given a Cardcomponent', () => {
  describe('when it is rendered', () => {
    let screen:any;
    beforeEach(() => {
      const card = {
        question: 'quest',
        answer: 'answer'
      };
      screen = render(<CardComponent card={card} />);
    });
    test('Then should match the snapshot', () => {
      expect(screen).toMatchSnapshot();
    });
  });
});
