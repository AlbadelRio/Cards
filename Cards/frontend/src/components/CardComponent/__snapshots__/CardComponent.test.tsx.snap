// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Cardcomponent when it is rendered Then should match the snapshot 1`] = `
<View>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "backgroundColor": "#EAECF2",
            "borderRadius": 3,
            "elevation": 10,
            "height": 200,
            "justifyContent": "center",
            "marginTop": 60,
            "padding": 10,
            "width": 350,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-end",
            }
          }
        />
        <Text>
          Question:
        </Text>
        <TextInput
          editable={false}
          onChangeText={[Function]}
          value="quest"
        />
        <Text>
          Answer:
        </Text>
        <TextInput
          editable={false}
          onChangeText={[Function]}
          value="answer"
        />
        <View
          style={
            Object {
              "alignItems": "flex-end",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View>
              <Image
                source={
                  Object {
                    "uri": "https://img.icons8.com/material-sharp/24/000000/edit--v3.png",
                  }
                }
                style={
                  Object {
                    "height": 30,
                    "width": 30,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
